lib/raty.js

```diff
@@ -0,0 +1,780 @@
+'use strict';
+//initial configuration
+const defaultoptions = Object.seal({
+  cancelButton: false,
+  cancelClass: 'raty-cancel',
+  cancelHint: 'Cancel this rating!',
+  cancelOff: 'cancel-off.png',
+  cancelOn: 'cancel-on.png',
+  cancelPlace: 'left',
+  click: undefined,
+  half: false,
+  halfShow: true,
+  hints: ['bad', 'poor', 'regular', 'good', 'gorgeous'],
+  iconRange: undefined,
+  iconRangeSame: false,
+  mouseout: undefined,
+  mouseover: undefined,
+  noRatedMsg: 'Not rated yet!',
+  number: 5,
+  numberMax: 20,
+  path: undefined,
+  precision: false,
+  readOnly: false,
+  round: { down: 0.25, full: 0.6, up: 0.76 },
+  score: undefined,
+  scoreName: 'score',
+  single: false,
+  space: true,
+  starHalf: 'star-half.png',
+  starOff: 'star-off.png',
+  starOn: 'star-on.png',
+  starType: 'img',
+  target: undefined,
+  targetFormat: '{score}',
+  targetKeep: false,
+  targetScore: undefined,
+  targetText: '',
+  targetType: 'hint',
+  initialize: true,
+});
+class Raty {
+  /**
+   *
+   * @param {object} options
+   * @param {string} element
+   *
+   */
+  constructor(element, options) {
+    //overwrite basic configuration
+    if (options) {
+      if (typeof options === 'object' && !Array.isArray(options)) {
+        this.opt = {
+          ...defaultoptions,
+          ...options,
+        };
+      } else {
+        throw new Error('Illegal parameter');
+      }
+    } else {
+      this.opt = defaultoptions;
+    }
+    this.self = document.querySelector(element);
+    this.element = element;
+
+    if (this.opt.initialize) {
+      this.init();
+    }
+  }
+
+  cancel(click) {
+    if (!this.opt.readOnly) {
```

Comment:
I think here we have an issue because `readOnly` is the config to start as it, but here we were checking the current state that can be or not be `readOnly`, so that's why we were using the `dataset`.